### PR TITLE
Add basic stability verification

### PR DESCRIPTION
**Pull request by mplantykow**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,31 +39,33 @@ ip a
...
-| **Argument**		| **Description**								| Default |
-| --------------------- | --------------------------------------------- 				| ------- |
-| --i			| Interface									| -	  |
-| --t			| Time of a single test								| 120	  |
-| --metric		| Evaluation metric (1 - MSE, 2 - RMSE, 3 - MAE)				| 1	  |
+| **Argument** | **Description**                                | Default 	|
+| ------------ | ---------------------------------------------- | --------- |
+| --i          | Interface                                      | -         |
+| --t          | Time of a single test                          | 120       |
+| --metric     | Evaluation metric (1 - MSE, 2 - RMSE, 3 - MAE) | 1         |
```
Comment (mplantykow) [inline README.md:67]: Should we move --metric to configureme.py?

File: main.py [hunk 2]
```diff
@@ -25,6 +25,30 @@ def __contains__(self, item):
...
+    return Kp, Ki
+
+def redefine_kp_ki_to_stable(Kp, Ki):
+    stable = False
+    while not stable:
+        Kp = Kp - config.reduction_determinant * Kp
+        Ki = Ki - config.reduction_determinant * Ki
+        if validate_stability(Kp, Ki):
+            stable = True
+    return Kp, Ki
```
Comment (mplantykow) [inline main.py:69]: Should I move 28-50 to a separate pyton file and include? Just 3 functions/25 lines, separate functionality on the other hand.

File: main.py [hunk 3]
```diff
@@ -109,10 +140,21 @@ def __iter__(self):
         y = x + 1
         for _ in range(confug.gen_num_inherited - x - 1):
             print(x, " + ", y)
-            new_generation.append(
-                Creature(population[sorted_scores_indexes[x]].Kp, population[sorted_scores_indexes[y]].Ki))
-            new_generation.append(
-                Creature(population[sorted_scores_indexes[y]].Kp, population[sorted_scores_indexes[x]].Ki))
+            if (config.stability_verification):
```
Comment (mplantykow) [inline main.py:205]: In this implementation, if a result of a crossover is not stable, we are slowly reducing kp and ki to make it fit to the equation (P+I)^2 < 4I. Is this ok? Maybe we should introduce a param that will let user pick wether we want to do that or just drop creature (like crossover_unstable_policy = [drop, reduce]).

File: main.py [hunk 4]
```diff
@@ -169,6 +215,8 @@ def __iter__(self):
         rand_y = random.uniform(-1, 1) #nosec
         new_ki = creature.Ki + (rand_y * config.gen_mutation_coef)
         new_ki = max(0, min(new_ki, config.gen_max_ki))
+        if not validate_stability(new_kp, new_ki):
```
Comment (mplantykow) [inline main.py:280]: Same as above:
In this implementation, if a result of a mutation is not stable, we are slowly reducing kp and ki to make it fit to the equation (P+I)^2 < 4I. Is this ok? Maybe we should introduce a param that will let user pick wether we want to do that or just drop creature (like mutation_unstable_policy = [drop, reduce]).

**Comment by mplantykow** [comment]
TBD LINTER

File: README.md [hunk 5]
```diff
@@ -39,31 +39,33 @@ ip a
...
-| --pop_size		| Number of epochs								| 10	  |
-| --max_kp		| Max value of Kp								| 15	  |
-| --max_kp		| Max value of Ki								| 15  	  |
-| --num_random		| Number of random parents added to each epoch					| 2	  |
-| --num_inherited	| Number of the best parents that are crossed to create a new generation	| 5	  |
-| --num_replicated	| Number of the best parents that are replicated to create a new generation 	| 4	  |
-| --mutation_coef	| Mutation coefficient								| 1	  |
-| --debug_level		| Determines level of debug prints						| 1	  |
-| --elite_size		| Number of elite chromosomes							| 1	  |
+Configureme.py file is used to configure the stability verification, genetic algorithm, and debug level.
```
Comment (mplantykow) [inline README.md:42]: Should I have it in a separate commit?

File: README.md (hunk 1 above)
Comment (MilenaOlech) [reply README.md:67]: Sounds good

File: main.py (hunk 3 above)
Comment (MilenaOlech) [reply main.py:205]: Could you please explain what is the background of this stability method?

File: main.py [hunk 6]
```diff
@@ -25,6 +25,32 @@ def __contains__(self, item):
     def __iter__(self):
         yield self
 
+def validate_stability(Kp, Ki):
+    if (((Kp + Ki)*(Kp + Ki)) < (4*Ki)):
```
Comment (MilenaOlech) [inline main.py]: What's the reason of adding this condition?

File: README.md (hunk 1 above)
Comment (MaciekMachni) [reply README.md:67]: Agree! :)

File: main.py (hunk 4 above)
Comment (MaciekMachni) [reply main.py:280]: Can we retry generating different random pair of parameters when they are not stable? I think if we drop them we can get in the deadlock where we'll not have any stable pair in a given mutation

File: main.py (hunk 6 above)
Comment (MaciekMachni) [reply main.py]: the goal is to skip unstable parameters and potentially replace them by a different pair

File: main.py (hunk 2 above)
Comment (MaciekMachni) [reply main.py:69]: You can - but maybe it'd be easier to create a function that will generate stable pair of Kp and Ki as an API for this file?

**Pull request merged**